clusterloader2/pkg/measurement/common/network/network_performance_measurement.go

```diff
@@ -0,0 +1,449 @@
+/*
+Copyright 2021 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// Package network captures network performance metrics
+// for protocol TCP,UDP,HTTP etc. The metrics are collected for baseline (1:1),
+// scale (N:M) pod ratios.Client and server pods located on different worker
+// nodes exchange traffic for specified time to measure the performance metrics.
+package network
+
+import (
+	"container/heap"
+	"context"
+	"fmt"
+	"sync"
+	"time"
+
+	"k8s.io/apimachinery/pkg/api/meta"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/client-go/dynamic"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/tools/cache"
+	"k8s.io/klog"
+	"k8s.io/perf-tests/clusterloader2/pkg/framework"
+	"k8s.io/perf-tests/clusterloader2/pkg/framework/client"
+	"k8s.io/perf-tests/clusterloader2/pkg/measurement"
+	measurementutil "k8s.io/perf-tests/clusterloader2/pkg/measurement/util"
+	"k8s.io/perf-tests/clusterloader2/pkg/util"
+)
+
+const (
+	podReadyTimeOut = 5 * time.Minute
+	// initialDelay in seconds after which pod starts sending traffic.
+	// Delay is used to synchronize all client pods to send traffic at same time.
+	initialDelayForTestExecution = 5 * time.Second
```

Comment:
Isn't 5s too low?

Reply:
We have tested upton 120-130 pods. It has worked without any issues. All pods were able to start within 5sec. We plan to increase it if required , once these TCs start running in actual cloud.

Reply:
This will be too low for running more clusters (and I would e.g. like to be able to run with 30 pods per node in 100-node cluster - so 3000 pods).

So we should bump that upfront already.